.github/workflows/python.yml

```diff
@@ -30,25 +32,124 @@ jobs:
         environment-file: .github/python_test_env.yml
 
     - name: Install
-      run: pip install . -v
+      run: python -m pip install . -v
 
     - name: Test
-      run: pytest --cov exactextract --cov-report lcov python/tests
+      run: python -m pytest --cov exactextract --cov-report lcov python/tests
 
     - name: Upload coverage report
       uses: codecov/codecov-action@v3
       with:
         files: coverage.lcov
         token: ${{ secrets.CODECOV_TOKEN }}
-      
+
     - name: Build source distribution
       if: matrix.os == 'ubuntu-latest' && matrix.python-version == '3.11'
       run: python -m build . --sdist
 
     - name: Upload distribution(s)
-      if: matrix.os == 'ubuntu-latest' && matrix.python-version == '3.11'
+      if: startsWith(github.event.ref, 'refs/tags') || github.event_name == 'release'
       uses: actions/upload-artifact@v3
       with:
         name: ${{ matrix.os }}-py${{ matrix.python-version }}
         path: ./dist/*
 
+  build_wheels:
+    name: Build ${{ matrix.arch }} wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    # if: startsWith(github.event.ref, 'refs/tags') || github.event_name == 'release'
```

Comment:
TODO: uncomment this before merging.

We don't need to build the wheels on all PR / Push